enzy_htp/structure/structure_region/capping.py

```diff
@@ -179,72 +289,271 @@ class CH3Cap(ResidueCap):
-        "HP21" : (0.360, -1.029,  0.000),
-        "HP22" : (0.360,  0.514,  0.891),
-        "HP23" : (0.360,  0.514, -0.891),
-        "CP1"  : (0.000,  0.000,  0.000),
-        "HP11" : (0.360, -1.029,  0.000),
-        "HP12" : (0.360,  0.514,  0.891),
-        "HP13" : (0.360,  0.514, -0.891),
-    }
-    """TODO improve this"""
+    """Maps the name of the link_atom to appropriate bond distance."""
+
+    def get_nterm_atoms(self) -> List[Atom]:
+        """Create the default n-terminal version of the CH3Cap with appropriate names."""
+        return [
+        Atom({'atom_name':"CP1" , 'x_coord':0.000, 'y_coord':  0.000, 'z_coord':  0.000, 'element_symbol':'C'}),
+        Atom({'atom_name':"HP11", 'x_coord':0.360, 'y_coord': -1.029, 'z_coord':  0.000, 'element_symbol':'H'}),
+        Atom({'atom_name':"HP12", 'x_coord':0.360, 'y_coord':  0.514, 'z_coord':  0.891, 'element_symbol':'H'}),
+        Atom({'atom_name':"HP13", 'x_coord':0.360, 'y_coord':  0.514, 'z_coord': -0.891, 'element_symbol':'H'}),]
+        
+    def get_cterm_atoms(self) -> List[Atom]:
+        """Create the default c-terminal version of the CH3Cap with appropriate names."""
+        return [
+        Atom({'atom_name':"CP2" , 'x_coord':0.000, 'y_coord':  0.000, 'z_coord':  0.000, 'element_symbol':'C'}),
+        Atom({'atom_name':"HP21", 'x_coord':0.360, 'y_coord': -1.029, 'z_coord':  0.000, 'element_symbol':'H'}),
+        Atom({'atom_name':"HP22", 'x_coord':0.360, 'y_coord':  0.514, 'z_coord':  0.891, 'element_symbol':'H'}),
+        Atom({'atom_name':"HP23", 'x_coord':0.360, 'y_coord':  0.514, 'z_coord': -0.891, 'element_symbol':'H'}),]
 
     @property
     def cap_type(self) -> str:
-        """hard coded cap type"""
+        """What type of """
         return "CH3"
 
     @property
     def net_charge(self) -> str:
         """hard coded net charge"""
         return 0
 
+    def anchor(self) -> None:
+        """Straightforward anchoring that is identical for both n-terminal and c-terminal capping."""
+        
+        p1 = np.array(self.link_atom.coord)
+        p2 = np.array(self.socket_atom.coord)
 
-class HCap(ResidueCap):
-    """the H cap
-    TODO move the default atom coord in?
-    TODO move bond distance in."""
-    FIXED_CHARGE_MAP = {
-        # C-ter
-        "HP21" : 0.0,
-        # N-ter
-        "HP11" : 0.0,
-    } # improve this
-    """fixed charge for a -H attached to the N/C-ter
-    of a residue."""
+        direction = p2 - p1
+        direction /= np.linalg.norm(direction)
+        rot_mat = rotation_matrix_from_vectors(np.array([1,0,0]), direction)
+        cap_bond_distance = self.CAP_BOND_DISTANCE[self.link_atom.name]
 
-    CAP_BOND_DISTANCE = {
-        "N" : 1.1,
-        "C" : 1.0,
+        for aa in self.atoms:
+            pt = np.array(aa.coord)
+            pt = np.transpose(np.matmul(rot_mat, np.transpose( pt  )))
+            pt += (direction*cap_bond_distance) + p1 
+            aa.coord = pt
```

Comment:
We should generalize it and put the math part in core/math_helper.py

Reply:
The current anchor dont work when the cap is not in the initial position I think. i.e.: not from 1,0,0
But it is needed in apply_to_geom

Reply:
This is true that the cap must be in the default position. However, EnzyHTP does not have a robust mechanism for doing internal coordinates, so we have to resort to hacky implementations like this. To be clear, I do not think we should have a robust mechanism for internal coordinates, but I don't think there is a better way to deal with it.

Reply:
I think it can be done with just cartesian coordinates. It is just transforming and rotating from different initial positions.

Reply:
I think I  misunderstood what you are saying. Yes you are right that the cap must "point" along 1,0,0. I thought this would be a good simplifying assumption to make the math easier, especially since there are only 1 or 2 capping groups even worth implementing after this